packages/datadog-instrumentations/src/redis.js

```diff
@@ -149,24 +134,28 @@ addHook({ name: 'redis', versions: ['>=0.12 <2.6'] }, redis => {
   return redis
 })
 
-function start (client, command, args, url = {}) {
-  const db = client.selected_db
-  const connectionOptions = client.connection_options || client.connection_option || client.connectionOption || url
-  startCh.publish({ db, command, args, connectionOptions })
+function getStartCtx (client, command, args, url = {}) {
+  return {
+    db: client.selected_db,
+    command,
+    args,
+    connectionOptions: client.connection_options || client.connection_option || client.connectionOption || url
+  }
 }
 
-function wrapCallback (finishCh, errorCh, callback) {
+function wrapCallback (finishCh, errorCh, ctx, callback) {
```

Comment:
It doesn't look like the parent context is restored in the callback. There is probably no test for this which is why everything is passing, but the callback should be wrapped with `finishCh.runStore` and a corresponding `bindFinish` handler on the subscriber side.